tinygrad/shape/shapetracker.py

```diff
@@ -79,6 +79,20 @@ def contiguous(self) -> bool: return len(self.views) == 1 and self.views[0].cont
   @property
   def shape(self) -> Tuple[sint, ...]: return self.views[-1].shape
 
+  @functools.cached_property
+  def var_vals(self) -> Dict[Variable, int]:
+    # Variable can come from shape or offset
```

Comment:
Do you have to do this in all the views, not just -1?

Reply:
i think you misread or i misunderstand, but i was doing on -1 only